src/modules/packages.c

```diff
@@ -68,6 +74,10 @@ void ffPrintPackages(FFinstance* instance)
         ffPrintError(instance, FF_PACKAGES_MODULE_NAME, 0, &instance->config.packagesKey, &instance->config.packagesFormat, FF_PACKAGES_NUM_FORMAT_ARGS, "No packages from known package managers found");
         return;
     }
+    FFstrbuf manjaroBranch;
+    ffStrbufInit(&manjaroBranch);
+    getActiveBranchManjaro(manjaroBranch);
+    ffStrbufRecalculateLength(&manjaroBranch); // needed for later length comparison...
```

Comment:
This is weird, you should't need it

Reply:
What is ?

Reply:
`ffStrbufRecalculateLength`

Reply:
The strbuf and placement is from your earlier suggestion, the Recalculate was needed to get length comparison to work.

Reply:
Yea i understood this, the thing is the length should always be up-to-date.

Reply:
In debug, it showed `manjaroBranch.length` to 0. If you want to test it, just create the file and use the contents I provided earlier.

Reply:
If i use the default file you provided, the length is expected to be 0. This happens when  `Branch =` was not found, for example commented out. This is why i set the default value when the file was found, but not the property.

Reply:
Tested it, works perfectly.

Reply:
Length surely shouldn't be 0 when the buffer is set with `ffStrbufSetS(&manjaroBranch, "stable");` ?

Reply:
That is right, and it is not 0 when i test it. Try with a breakpoint a bit later in the code.

Reply:
Removing that line makes the branch info disappear. IDK what is happening.

Reply:
Compiler optimizations probably as i force -O3 in CMakeLists.txt. What does a printf say about the length?

Reply:
```c
        if(pacman > 0)
        {
            printf("%u (pacman)", pacman);
            printf("Len: %d", manjaroBranch.length);
            if(manjaroBranch.length > 0)
                printf("[%s]", manjaroBranch.chars);
            if((all = all - pacman) > 0)
                printf(", ");
        };
```
`Packages: 1499 (pacman)Len: 0, 1 (flatpak)`

Reply:
Try with %u

Reply:
same

Reply:
Did you rewrite anything at all before testing?

Reply:
Did some changes to ffGetPropValue, but they should't affect you. They are already pushed so you can just pull them.

Reply:
Pulled 3 hours ago

Reply:
I've tested with the file not existing, it existing with branch as a comment and with a changed branch. Works all time:
```c
FFstrbuf manjaroBranch;
ffStrbufInit(&manjaroBranch);

if(ffParsePropFile("/etc/pacman-mirrors.conf", "Branch =", &manjaroBranch) && manjaroBranch.length == 0)
    ffStrbufSetS(&manjaroBranch, "stable");

if(instance->config.packagesFormat.length == 0)
{
    ffPrintLogoAndKey(instance, FF_PACKAGES_MODULE_NAME, 0, &instance->config.batteryKey);

    #define FF_PRINT_PACKAGE(name) \
    if(name > 0) \
    { \
        printf("%u ("#name")", name); \
        if((all = all - name) > 0) \
            printf(", "); \
    };

    if(pacman > 0)
    {
        printf("%u (pacman)", pacman);
        if(manjaroBranch.length > 0)
            printf("[%s]", manjaroBranch.chars);
        if((all = all - pacman) > 0)
            printf(", ");
    };
```

Reply:
You did rewrite it...

Reply:
I just optimized the ffStrbufTrim, like i said it should't change how the method works.

Reply:
Sure enough, now it works. Length was somehow lost using the `getActiveBranchManjaro()` function.
Should it not persist?

Reply:
Ah now i see your error, you have to pass FFstrbuf as a pointer to persist length. I didn't look at this because i just commented that the function is not needed. Only the data of a FFstrbuf is stored on ram.

Reply:
Well I did say "anything at all", Linus. ;)

Reply:
My fault :)